File: leetcode/1768.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {

        int fullSize = word1.size() + word2.size();
        
        std::string mergedString = "";

        int countIndex = 0;

        while( fullSize != mergedString.size() ) {

            if( word1.size() > countIndex) {
                mergedString += word1[countIndex];
            }

            if( word2.size() > countIndex ) {
                mergedString += word2[countIndex];
            }

            countIndex++;
        } 
    
        return mergedString;
    }
};

File: leetcode/1768.ts
function mergeAlternately(word1: string, word2: string): string {

    const wordLength: number = word1.length + word2.length;

    let mergedString = "";

    let countIndex: number = 0;

    while( mergedString.length !== wordLength) {

        if(word1[countIndex]) {
            mergedString = mergedString + word1[countIndex];
        }

        if(word2[countIndex]) {
            mergedString = mergedString + word2[countIndex];
        }

        countIndex++;
    }

    return mergedString;
};